tests/imapp_benchmark/imapp_bench.go

```diff
@@ -0,0 +1,249 @@
+
+	// Initialize some constant calldata of 128KB, 2^17 bytes.
+	// This means, if we offset between 0th and 2^16th byte, we can fetch between 0 and 2^16 bytes (64KB)
+	// In consequence, we need args to memory-copying OPCODEs to be between 0 and 2^16, 2^16 fits in a PUSH3,
+	// which we'll be using to generate arguments for those OPCODEs.
+	calldata = []byte(strings.Repeat("{", 1<<17))
+
+	// Warm-up. **NOTE** we're keeping tracing on during warm-up, otherwise measurements are off
+	cfg.EVMConfig.Debug = false
+	_, _, errWarmUp := runtime.Execute(bytecode, calldata, cfg)
+
+	if errWarmUp != nil {
+		b.Error()
+		return
+	}
+
+	b.ResetTimer()
+
+	var exBytes []byte
+
+	for i := 0; i < b.N; i++ {
+		exBytes2, _, err := runtime.Execute(bytecode, calldata, cfg)
+		exBytes = exBytes2
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+	}
+
+	result = exBytes
+}
+
+func BenchmarkBytecodeExecutionIndividualWithWarmup(b *testing.B) {
+	var calldata []byte
+	var exBytes []byte
+
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+
+		cfg := new(runtime.Config)
+		setDefaults(cfg)
+		// from `github.com/ethereum/go-ethereum/core/vm/runtime/runtime.go:109`
+		cfg.State, _ = state.New(common.Hash{}, state.NewDatabase(rawdb.NewMemoryDatabase()), nil)
+
+		// Initialize some constant calldata of 128KB, 2^17 bytes.
+		// This means, if we offset between 0th and 2^16th byte, we can fetch between 0 and 2^16 bytes (64KB)
+		// In consequence, we need args to memory-copying OPCODEs to be between 0 and 2^16, 2^16 fits in a PUSH3,
+		// which we'll be using to generate arguments for those OPCODEs.
+		calldata = []byte(strings.Repeat("{", 1<<17))
+
+		// Warm-up. **NOTE** we're keeping tracing on during warm-up, otherwise measurements are off
+		cfg.EVMConfig.Debug = false
+
+		b.StartTimer()
+
+		exBytes2, _, err := runtime.Execute(bytecode, calldata, cfg)
+		exBytes = exBytes2
+
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+	}
+
+	result = exBytes
+}
+
+func BenchmarkBytecodeExecutionIndividual(b *testing.B) {
+	var calldata []byte
+	var exBytes []byte
+
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+
+		cfg := new(runtime.Config)
+		setDefaults(cfg)
+		// from `github.com/ethereum/go-ethereum/core/vm/runtime/runtime.go:109`
+		cfg.State, _ = state.New(common.Hash{}, state.NewDatabase(rawdb.NewMemoryDatabase()), nil)
+
+		// Initialize some constant calldata of 128KB, 2^17 bytes.
+		// This means, if we offset between 0th and 2^16th byte, we can fetch between 0 and 2^16 bytes (64KB)
+		// In consequence, we need args to memory-copying OPCODEs to be between 0 and 2^16, 2^16 fits in a PUSH3,
+		// which we'll be using to generate arguments for those OPCODEs.
+		calldata = []byte(strings.Repeat("{", 1<<17))
+
+		// Warm-up. **NOTE** we're keeping tracing on during warm-up, otherwise measurements are off
+		cfg.EVMConfig.Debug = false
+		_, _, errWarmUp := runtime.Execute(bytecode, calldata, cfg)
```

Comment:
do we want to keep measuring this variant at all, given that we're using `bench`? I'd keep it simple and drop it completely

Reply:
Dropped